src/pip/_internal/commands/install.py

```diff
@@ -222,6 +222,14 @@ def add_options(self) -> None:
             default=True,
             help="Do not warn about broken dependencies",
         )
+        self.cmd_opts.add_option(
+            "--no-warn-when-using-as-a-root-user-to-manage-os-packages",
+            dest="no_warn_about_root_user",
+            action="store_true",
+            help="Disable warning when there are legitimate uses "
+            "for managing os packages as root user"
+            "(for example in some cases of container builds)",
```

Comment:
```suggestion
            help="Do not warn when used as a root user",
```

I don't think we should elaborate on the use case in the help string (see other help strings above, for how terse these strings usually are).

Reply:
Fine for me.